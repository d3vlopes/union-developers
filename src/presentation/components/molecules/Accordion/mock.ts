import { AccordionProps } from '.'

export const accordionMock: AccordionProps = {
  title: 'O que é o Union?',
  content: `
    <p>O Union é um ambiente prático e imersivo próximo do encontrado nas empresas, desafiando seus membros a resolver problemas reais, com cronogramas e prazos de entrega.</p> 
    <p>No Union, você desenvolve as skills mais procuradas pelo mercado, trabalha com ferramentas e metodologias utilizada pelas empresas, além de ganhar experiência trabalhando com outras pessoas de varias partes do Brasil.</p> 
    <p>Você vai vivenciar na prática como é o dia a dia de um desenvolver em uma empresa e ficar mais preparado para conquistar sua vaga.</p>
  `,
}
