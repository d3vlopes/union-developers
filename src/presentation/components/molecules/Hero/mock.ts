import { HeroProps } from './types'

type MockType = 'default' | 'long-title' | 'long-description'

const defaultProps: HeroProps = {
  title:
    '<h1>Quer se tornar um desenvolvedor <strong>diferenciado?</strong></h1>',
  description:
    'Aqui você vivencia na prática como é o dia a dia de um desenvolvedor em uma empresa e se prepara para enfrentar os desafios do mercado de forma prática e imersiva!',
  buttonText: 'Quero participar',
  buttonTarget: '#join',
  backgroundMobile:
    'https://media.graphassets.com/j1rOxE97RsCbeyFrMU4z',
  backgroundDesktop:
    'https://media.graphassets.com/66rHD5NBR0K9lcIF9t62',
  logoUrl: 'https://media.graphassets.com/zRjyxnD5RyGdG8xVpll8',
  logoAlt: 'Description logo',
}

export const heroMockFactory: Record<MockType, HeroProps> = {
  default: {
    ...defaultProps,
  },
  'long-title': {
    ...defaultProps,
    title:
      '<h1>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!</h1>',
  },
  'long-description': {
    ...defaultProps,
    description: `
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Placeat, alias consequatur. Totam vero omnis praesentium, repellat dicta veniam illum, unde a sed doloribus quis non dolores autem ab nulla neque!
    `,
  },
}
