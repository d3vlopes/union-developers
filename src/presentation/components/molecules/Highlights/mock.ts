import { HighlightModel } from '@/domain/models'

export const highlightsMock: HighlightModel[] = [
  {
    id: '1',
    icon: {
      url: 'https://media.graphassets.com/qssZQEUSEaAQoM6oznCw',
      alt: 'rocket',
    },
    title: 'Crescimento profissional',
    description:
      'Eleve o seu conhecimento, aperfeiçoe suas habilidades e alavanque o seu potencial com uma experiência intensa que te ajudará a ser um profissional ainda melhor.',
  },
  {
    id: '2',
    icon: {
      url: 'https://media.graphassets.com/BN50QqvcR67hdjNyuqkA',
      alt: 'pencil and scale',
    },
    title: 'Aumente sua caixa de ferramentas',
    description:
      'Descubra novas metodologias e encare desafios que agregam em seu conhecimento profissional, essenciais para avançar na sua trajetória.',
  },
  {
    id: '3',
    icon: {
      url: 'https://media.graphassets.com/WR1yF2qGRmm8jWLCyw0D',
      alt: 'like',
    },
    title: 'Problemas Reais',
    description:
      'Trabalhe em projetos reais com equipes reais para sair da sua zona de conforto e acelerar seu aprendizado.',
  },
  {
    id: '4',
    icon: {
      url: 'https://media.graphassets.com/2S5Pdk5oShOJUlIz8p7X',
      alt: 'planet',
    },
    title: 'Ambiente Imersivo',
    description:
      'Espaço completo que te prepara, conecta com o mercado e permite absorver conteúdos fundamentais para seguir a sua jornada profissional.',
  },
  {
    id: '5',
    icon: {
      url: 'https://media.graphassets.com/KHCFnl40RyKQqOx6Bjuz',
      alt: 'bulb on',
    },
    title: 'Experiência',
    description:
      'Com os processos fundamentais e domínio das ferramentas te dá a experiência necessária para tomar as melhores decisões no mundo real.',
  },
  {
    id: '6',
    icon: {
      url: 'https://media.graphassets.com/XH5lk8IJQ1Cz1M2AE1Lu',
      alt: 'game controller',
    },
    title: 'Vivência na Prática',
    description:
      'Vivencie o dia a dia do mercado com um time de diferentes mentalidades para solucionar problemas em diversas áreas.',
  },
  {
    id: '7',
    icon: {
      url: 'https://media.graphassets.com/eBMuKBvTLulsEbZySGwt',
      alt: 'star',
    },
    title: 'Desenvolvimento de Skills',
    description:
      'Aprenda a comunicar suas ideias em um ambiente profissional e melhorar suas habilidades de gerenciamento de tempo para se destacar em sua carreira.',
  },
  {
    id: '8',
    icon: {
      url: 'https://media.graphassets.com/sanArFzbQUipTnjRVl78',
      alt: 'person',
    },
    title: 'Mentoria',
    description:
      'No Union você irá ter do lado um profissional experiente que vai te guiar e mostrar os atalhos para que você consiga evoluir mais rápido em menos tempo.',
  },
]
