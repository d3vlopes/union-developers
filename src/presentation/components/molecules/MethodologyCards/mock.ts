import { MethodologyModel } from '@/domain/models'

export const methodologyCardsMock: MethodologyModel[] = [
  {
    id: '1',
    title: 'Ativa',
    description:
      'É uma metodologia de aprendizado que coloca o aluno como protagonista, onde ele é o responsável por ir atrás dos conhecimentos necessários, ficando o professor como um mentor, que vai mostrar o caminho e indicar os melhores conteúdos, que vai guiar o estudo e os métodos, mas também deixar espaço para o aluno explorar.',
  },
  {
    id: '2',
    title: 'Andragogia',
    description:
      'É uma metodologia de ensino para adultos, que tem mais liberdade e colaboração, permitindo que o aluno decida quando, como e o que estudar, tornando o processo de aprendizagem mais sólido e eficaz. É uma metodologia de ensino mais prática, que estimula o aluno a resolver problemas reais, que estão relacionado ao seu interesse.',
  },
  {
    id: '3',
    title: 'Ágeis',
    description:
      'A metodologia ágil é uma metodologia que busca entregar valor a todo momento, estimula uma maior colaboração entre o time, permitindo que tenha uma visão mais transparente, tendo liberdade para fazer escolhas e tomar decisões, também estimula a melhora continua entre os membros, processos e da qualidade dos entregáveis.',
  },
]
