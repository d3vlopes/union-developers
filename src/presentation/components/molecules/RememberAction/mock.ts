import { RememberActionProps } from '.'

export const rememberActionMock: RememberActionProps = {
  title:
    'Vivencia na prática, ambiente imersivo, desafios e muito mais!',
  subtitle: 'Inscreva-se já! Vagas limitadas',
  buttonText: 'Inscrever-se agora',
  buttonTarget: '#subscribe',
  logoUrl: 'https://media.graphassets.com/WayysMB8RQyrgcahBMQs',
  logoAlt: 'Logo description',
}
