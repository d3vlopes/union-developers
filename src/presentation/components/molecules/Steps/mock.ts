import { StepModel } from '@/domain/models'

export const stepsMock: StepModel[] = [
  {
    id: '1',
    title: 'Pré-inscrição',
    description:
      'Inicie a sua inscrição preenchendo o nosso formulário',
    number: 1,
  },
  {
    id: '4',
    title: 'Entrevista',
    description:
      'Nessa fase iremos bater um papo para que possamos conhecer melhor você',
    number: 4,
  },
  {
    id: '3',
    title: 'Desafio técnico',
    description: 'Após a análise você receberá um desafio prático',
    number: 3,
  },
  {
    id: '2',
    title: 'Análise do formulário',
    description: 'O nosso time irá analisar suas informações',
    number: 2,
  },
]
