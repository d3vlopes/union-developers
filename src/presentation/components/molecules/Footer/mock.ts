import { generateId } from '@/utils'

import { logoMock } from '@/presentation/components/atoms/Logo/mock'

import { FooterProps } from '.'

export const footerMock: FooterProps = {
  logo: {
    imgUrl: logoMock.imgUrl,
    alt: logoMock.alt,
  },
  socialLinks: [
    {
      id: generateId(),
      icon: {
        imgUrl: 'https://media.graphassets.com/4AjPcOSSuK4kIKUv8Mdg',
        alt: 'Um ícone de envelope representando o email do Union',
      },
      label: 'Email',
      target: 'contato.uniongroup@gmail.com',
    },
    {
      id: generateId(),
      icon: {
        imgUrl: 'https://media.graphassets.com/y4KMQdygSH6uZx46pxjU',
        alt: 'Ícone com as letras in representando a logo do Linkedin',
      },
      label: 'Linkedin',
      target: 'www.linkedin.com/in/company/uniondevelopers',
    },
    {
      id: generateId(),
      icon: {
        imgUrl: 'https://media.graphassets.com/4lDgbi8FQpq8QYSiranx',
        alt: 'Um gatinho representando o logo do Github',
      },
      label: 'Github',
      target: 'www.github.com/union-group',
    },
  ],
  companyName: 'Union Developers',
  companySlogan:
    'O Union proporciona experiências para desenvolvedores em inicio de carreira, desafiando seus membros a resolver problemas reais.',
  navigationMenuLinks: [
    {
      id: generateId(),
      title: 'Saiba mais',
      links: [
        {
          label: 'Sobre',
          target: '#about',
        },
        {
          label: 'Depoimentos',
          target: '#testimonials',
        },
      ],
    },
    {
      id: generateId(),
      title: 'Dúvidas',
      links: [
        {
          label: 'FAQ',
          target: '#questions',
        },
        {
          label: 'Como funciona',
          target: '#steps',
        },
      ],
    },
  ],
}
