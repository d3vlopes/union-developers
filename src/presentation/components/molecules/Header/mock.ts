import { logoMock } from '@/presentation/components/atoms/Logo/mock'

import { HeaderProps } from './types'

export const headerMock: HeaderProps = {
  logoUrl: logoMock.imgUrl,
  logoAlt: logoMock.alt,
  navigationLinks: [
    {
      label: 'Sobre',
      target: '#about',
    },
    {
      label: 'Benefícios',
      target: '#highlights',
    },
    {
      label: 'Etapas',
      target: '#steps',
    },
    {
      label: 'Dúvidas',
      target: '#questions',
    },
  ],
  buttonText: 'Junte-se a nós',
  buttonTarget: '#join',
}
