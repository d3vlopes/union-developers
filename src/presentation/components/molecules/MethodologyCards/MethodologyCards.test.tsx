import { render, screen } from '@/__tests__/helpers'

import { methodologyCardsMock } from './mock'

import { MethodologyCards } from '.'

const { getByRole, getByText, getAllByRole } = screen

const cardMock = methodologyCardsMock[0]

describe('<MethodologyCards />', () => {
  it('should render heading', () => {
    render(<MethodologyCards items={methodologyCardsMock} />)

    const heading = getByRole('heading', {
      name: cardMock.title,
    })

    expect(heading).toBeInTheDocument()
  })

  it('should render description', () => {
    render(<MethodologyCards items={methodologyCardsMock} />)

    const description = getByText(cardMock.description)

    expect(description).toBeInTheDocument()
  })

  it('should render many cards', () => {
    render(<MethodologyCards items={methodologyCardsMock} />)

    const cardTitles = getAllByRole('heading')

    expect(cardTitles).toHaveLength(methodologyCardsMock.length)
  })

  it('should match snapshot', () => {
    const { container } = render(
      <MethodologyCards items={methodologyCardsMock} />,
    )

    expect(container).toMatchSnapshot()
  })
})
