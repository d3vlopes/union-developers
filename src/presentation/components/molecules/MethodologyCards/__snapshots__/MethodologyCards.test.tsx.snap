// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MethodologyCards /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 34.1rem;
}

.c1 .c2 {
  color: #E2E2E2;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem 5.6rem;
  border-radius: 10px 10px 0 0;
  background: linear-gradient(225deg,#0048FE 0%,#851D86 100%);
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
}

.c3 h3 {
  font-size: 2.4rem;
  font-weight: 700;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.6rem;
  border-radius: 0 0 10px 10px;
  background: linear-gradient( 225deg, rgba(0,72,254,0.1) 0%, rgba(133,29,134,0.1) 100% );
}

.c4 p {
  font-size: 1.6rem;
  color: #E2E2E2;
  line-height: 150%;
  text-align: center;
}

@media (min-width:768px) {
  .c0 {
    gap: 3.2rem;
  }
}

@media (min-width:1400px) {
  .c0 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

@media (min-width:768px) {
  .c1:hover {
    box-shadow: 0px 0px 60px 0px rgba(133,29,134,0.7);
  }

  .c1:hover .c2 {
    background: #FBFBFB;
    color: #282828;
  }
}

@media (min-width:1024px) {
  .c1 {
    max-width: 39.7rem;
  }
}

@media (min-width:1024px) {
  .c3 {
    padding: 4rem 9.2rem;
  }

  .c3 h3 {
    font-size: 3.2rem;
  }
}

@media (min-width:1024px) {
  .c4 {
    padding: 1.8rem;
  }

  .c4 p {
    font-size: 1.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2 c3"
      >
        <h3>
          Ativa
        </h3>
      </header>
      <div
        class="c4"
      >
        <p>
          Metodologia ativa é uma abordagem educacional que coloca o aluno como protagonista, onde ele é o responsável por ir atrás dos conhecimentos necessários, ficando o professor como um mentor, que vai mostrar o caminho para que ele possa ter sucesso no aprendizado.
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <header
        class="c2 c3"
      >
        <h3>
          Andragogia
        </h3>
      </header>
      <div
        class="c4"
      >
        <p>
          É uma metodologia de ensino para adultos, que tem mais liberdade e colaboração, permitindo que o aluno decida quando, como e o que estudar, tornando o processo de aprendizagem mais sólido e eficaz, a andragogia valoriza a autonomia e a experiência dos adultos como base para a aprendizagem 
        </p>
      </div>
    </div>
    <div
      class="c1"
    >
      <header
        class="c2 c3"
      >
        <h3>
          Ágeis
        </h3>
      </header>
      <div
        class="c4"
      >
        <p>
          A metodologia ágil é uma metodologia que estimula a colaboração entre o time, permitindo que tenha uma visão mais transparente, tendo liberdade para fazer escolhas e tomar decisões, também estimula a melhora continua entre os membros, processos e da qualidade geral do trabalho.
        </p>
      </div>
    </div>
  </div>
</div>
`;
